base/reflection.jl

```diff
@@ -627,7 +627,9 @@ true
 !!! compat "Julia 1.5"
     This function requires at least Julia 1.5.
 """
-ismutable(@nospecialize(x)) = (@_total_meta; typeof(x).name.flags & 0x2 == 0x2)
+ismutable(@nospecialize(x)) = (@_total_meta; (typeof(x).name::Core.TypeName).flags & 0x2 == 0x2)
```

Comment:
perhaps this?

```suggestion
ismutable(@nospecialize(x)) = (@_total_meta; (Base.inferencebarrier(typeof(x))::DataType).name.flags & 0x2 == 0x2)
```

(though we really need to work on teaching inference to do this itself)

Reply:
I get 

```
error during bootstrap:
LoadError(at "compiler/compiler.jl" line 3: LoadError(at "compiler/validation.jl" line 4: UndefVarError(var=:Base, scope=Core.Compiler)))
```

when doing this locally and running `make`.

Reply:
I implemented the corrected version in [3f5436c](https://github.com/JuliaLang/julia/pull/52170/commits/3f5436c633943ceb0f101022a6b3952e3fb3f3f6)